storage/prometheus/scanners.go

```diff
@@ -53,6 +53,12 @@ func (p Scanners) NewVectorSelector(
 	hints storage.SelectHints,
 	logicalNode logicalplan.VectorSelector,
 ) (model.VectorOperator, error) {
+	// Update hints with projection information if available
+	if logicalNode.Projection != nil {
```

Comment:
https://github.com/prometheus/prometheus/pull/16423 you can use projection hints now

Reply:
Done using the new prometheus